test/integration/apiserver/flowcontrol/fight_test.go

```diff
@@ -0,0 +1,237 @@
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	"k8s.io/klog/v2"
+)
+
+const timeFmt = "2006-01-02T15:04:05.999"
+
+func TestConfigConsumerFight(t *testing.T) {
+	// Disable the APF FeatureGate so that the normal config consumer
+	// controller does not interfere
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.APIPriorityAndFairness, false)()
+	_, loopbackConfig, closeFn := setup(t, 100, 100)
+	defer closeFn()
+	ctx := context.Background()
+	const size = 3
+	stopCh := make(chan struct{})
+	now := time.Now()
+	clk := clock.NewFakeClock(now)
+	ctlrs := map[bool][]utilfc.TestableInterface{
+		false: make([]utilfc.TestableInterface, size),
+		true:  make([]utilfc.TestableInterface, size)}
+	// maps FS key -> last written ResourceVersion
+	var lastRVs map[string]string = nil
+	var rvMutex sync.Mutex // hold when accessing notifiedRVs
+	// maps invert -> i -> FS key -> last notified ResourceVersion
+	notifiedRVs := map[bool][]map[string]string{
+		false: make([]map[string]string, size),
+		true:  make([]map[string]string, size)}
+	foreach := func(visit func(invert bool, i int)) {
+		for i := 0; i < size; i++ {
+			// The order of the following iteration is not deterministic,
+			// and that is good.
+			for invert := range ctlrs {
+				visit(invert, i)
+			}
+		}
+	}
+	foreach(func(invert bool, i int) {
+		rvMutex.Lock()
+		defer rvMutex.Unlock()
+		notifiedRVs[invert][i] = map[string]string{}
+	})
+	foreach(func(invert bool, i int) {
+		myConfig := rest.CopyConfig(loopbackConfig)
+		myConfig = rest.AddUserAgent(myConfig, fmt.Sprintf("invert=%v, i=%d", invert, i))
+		myClientset := clientset.NewForConfigOrDie(myConfig)
+		fcIfc := myClientset.FlowcontrolV1beta1()
+		fsIfc := fcIfc.FlowSchemas()
+		if lastRVs == nil {
+			lastRVs = make(map[string]string)
+			// Wait until every FlowSchema is defined, and record its RV
+			allFlowSchemas := append(fcboot.MandatoryFlowSchemas, fcboot.SuggestedFlowSchemas...)
+			err := wait.Poll(time.Second, wait.ForeverTestTimeout, func() (bool, error) {
+				for _, fs := range allFlowSchemas {
+					fs2, err := fsIfc.Get(ctx, fs.Name, metav1.GetOptions{})
+					if err != nil {
+						return false, err
+					}
+					t.Logf("Got initial FlowSchema %#+v", fs2)
+					key, _ := cache.MetaNamespaceKeyFunc(fs2)
+					lastRVs[key] = fs2.ResourceVersion
+				}
+				return true, nil
+			})
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+		informerFactory := informers.NewSharedInformerFactory(myClientset, 0)
+		fieldMgr := utilfc.ConfigConsumerAsFieldManager
+		foundToDangling := func(found bool) bool { return !found }
+		if invert {
+			fieldMgr = fieldMgr + "x"
+			foundToDangling = func(found bool) bool { return found }
+		}
+		ctlr := utilfc.NewTestable(utilfc.TestableConfig{
+			Name:  fmt.Sprintf("Controller%d[invert=%v]", i, invert),
+			Clock: clk,
+			FinishHandlingNotification: func(wq workqueue.RateLimitingInterface, obj interface{}) {
```

Comment:
this doesn't add to the workqueue like the normal one?  That seems a little odd.